content/blog/2019-01-07-we-mastered-the-art-of-programming.md

```diff
@@ -0,0 +1,63 @@
+---
+title: We Mastered the Art of Programming
+author: Jean-Thomas Baillargeon
+date: '2019-01-07'
+slug: we-mastered-the-art-of-programming
+type: post
+tags:
+- software engineering
+- best practices
+description: ""
+featured: "we-mastered-the-art-of-programming-cover.jpeg"
+featuredpath: "img/headers/"
+---
+
+This is probably going to sound cliché and trivial but I just realised that I learned how to use a computer before learning to use a pen. I launched my favorite game on a MS/DOS terminal a few years before writing my name on paper.
+
+This is cliché and trivial since this holds true for many of us data scientist growing up with a computer. We found a way to download music out of napster, limewire or torrents - whatever was working at that time. We figured how to burn our friends’ game on a CD-ROM and then crack it with a fake CD KEY. We learned how to repair the family’s printer. We understood how to connect our laptop to Eduroam. We made very serious and scientific calculations for school points. We then moved to a workplace where we grasp the depth of the `vLookup`. We connected thousands of *Excel* spreadsheets together. We mastered the art of programming.
+
+## problem solving at it's finest.
+
+What is connecting the dot between all this, is the curiosity and the satisfaction of solving a problem, a computer problem. We make things work. This is how we get points on a school assignment. This is how our boss sees us as the achiever. This is how we create pure value. Based on this, it only makes sense that the faster you solve a computer problem, the better you are.
+
+But computer problem seldom get stashed away once solved. They come back to haunt you and require adjustments to the code so it fits the new constraints. Great a new problem! You have to find the fastest way to hack the code. This is fun until one or two things happen: the code you have to modify comes from a *bad* programmer or something breaks every time you modify something else. It will eventually work, but it’ll take a little longer than the previous iteration. This continues, you realise that the *bad* programmer is your past self from a few month ago and you fear modifying anything since you have no faith this [*Jenga*](https://secure.img1-fg.wfcdn.com/im/93997415/resize-h800%5Ecompr-r85/4885/48852016/Jenga%25AE+giant%25u2122+premium+jeu+de+bois+franc.jpg) will hold at all.
```

Comment:
Great, (virgule)

Reply:
This is great until one *of* two things ?

Reply:
Modify sounds weird : alter, change, revise ?

Reply:
A few month*s* ago

Reply:
xie xie fixed